website/server/controllers/api-v3/chat.js

```diff
@@ -101,6 +111,22 @@ api.postChat = {
     let lastClientMsg = req.query.previousMsg;
     chatUpdated = lastClientMsg && group.chat && group.chat[0] && group.chat[0].id !== lastClientMsg ? true : false;
 
+    // Check for spam by seeing if the current user has posted too many messages in a short window
+    if (!(user.contributor.level >= SPAM_CONTRIBUTOR_LEVEL) && (group.type === 'party' || group.type === 'guild')) {
+      let currentTime = Number(new Date());
+      let userMessages = 0;
+      for (let i = 0; i < group.chat.length; i++) {
+        let message = group.chat[i];
+        if (message.uuid === user._id && currentTime - message.timestamp <= SPAM_WINDOW_LENGTH) {
+          if (++userMessages >= SPAM_MESSAGE_LIMIT) {
+            throw new NotFound(res.t('messageGroupChatSpam'));
```

Comment:
yeah not found doesn't really make sense, I had just done that because it seemed to me like a similar situation as chat privileges being revoked that did use a 404 not found. Would a 403 be more appropriate for the application? Is that one that is handled in the code base well enough?